scripts/genlocaledata.js

```diff
@@ -0,0 +1,225 @@
+    console.log(
+        "Usage: genlocaledata.js [-i ilibPath] [-l locales] [-o outDir] \n" +
+        "Generate the ilib locale data js files.\n\n" +
+        "-h or --help\n" +
+        "  this help\n" +
+        "-i or --ilibPath\n" +
+        "  Location where ilib is installed. (Default: '.') \n" +
+        "-l or --locales\n" +
+        "  Restrict operation to only the given locales. Locales should be given as \n" +
+        "  a comma-separated list of BCP-47 locale specs. By default it will operate\n"+
+        "  with all locales that the webOS Platform needs to support.\n" +
+        "-o or --outDir\n" +
+        "  Directory to place the [language].js. (Default: './tmp')\n" +
+        "--minify\n" +
+        "  Minify the generated js files. (Default: 'false')");
+    process.exit(1);
+}
+
+process.argv.forEach(function (val, index, array) {
+    if (val === "-h" || val === "--help") {
+        usage();
+    }
+});
+
+if (process.argv.length < 2) {
+    console.error('Error: not enough arguments');
+    usage();
+}
+var ilibPath = ".";
+var outDir = "./tmp";
+var locales
+var isMinify = false;
+var argv = process.argv;
+for(var i=0; i < argv.length;i++){
+    var val = argv[i];
+    if (val === "-h" || val === "--help") {
+        settings.help = true;
+    } else if (val === "-l" || val === "--locales") {
+        if (i < argv.length && argv[i+1]) {
+            locales = argv[++i].split(",");
+        }
+    } else if (val === "-i" || val === "--ilibPath") {
+        ilibPath = argv[++i];
+    } else if (val === "-o" || val === "--outDir") {
+        outDir = argv[++i];
+    } else if (val === "--minify") {
+        isMinify = true;
+    } else {
+        console.log("wrong options");
+    }
+}
+
+var iliblocalePath = path.join(ilibPath, "js/data/locale");
+var ilibincPath = path.join(ilibPath, "js/lib/ilib-standard-flutter-inc.js");
+var localeList = locales ? locales : defaultlocales;
+
+console.log("genlocaledata - generate locale data file.\n" +
+    "Copyright (c) 2024 JEDLSoft");
+
+var dataMap = {
+    "PhoneFmt.js": ["phonefmt"],
+    "DateRngFmt.js": ["dateformats", "sysres"],
+    "ListFmt.js": ["list"],
+    "LocaleMatcher.js": ["localematch"],
+    "PhoneLocale.js": ["phoneloc"],
+    "AddressFmt.js": ["address", "addressres", "regionnames"],
+    "PhoneGeoLocator.js": ["iddarea", "area", "extarea", "extstates", "phoneres"],
+    "NumberingPlan.js": ["numplan"],
+    "Name.js": ["name"],
+    "Currency.js": ["currency"],
+    "DateFmt.js": ["dateformats", "sysres"],
+    "TimeZone.js": ["localeinfo"], // except zoneinfo
+    "LocaleInfo.js": ["localeinfo"],
+    "Address.js":["address", "countries", "nativecountries", "ctrynames"],
+    "NameFmt.js": ["name"],
+    "NumFmt.js.js": ["localeinfo", "currency"],
+    "ResBundle.js": ["pseudomap"],
+    "Country.js": ["ctryreverse"],
+    "PhoneNumber.js": ["states", "idd", "mnc"],
+    "IString.js": ["plurals"],
+    "DurationFmt.js": ["dateformats, sysres"],
```

Comment:
Need separate strings here

Reply:
I have corrected all comments in the `genlocaledata.js` file, as we discussed in the meeting. The tool code is temporirily. It will be moved to the ilib-assemble repository and re-managed.